src/cli/cli_joiner.hpp

```diff
@@ -79,6 +79,7 @@ class Joiner
         otError (Joiner::*mCommand)(uint8_t aArgsLength, char *aArgs[]);
     };
 
+    otError ProcessAdvertisement(uint8_t aArgsLength, char *aArgs[]);
```

Comment:
Maybe I don't have the context of why we want to introduce these APIs, but please note that the Joiner Advertisement "Set" APIs are available even when the Joiner flag is not enabled. That means they are general APIs for any roles but not only MeshCOP Joiners. Do we want align the scenarios between "Get" and "Set" APIs?

Reply:
>  the Joiner Advertisement "Set" APIs are available even when the Joiner flag is not enabled. That means they are general APIs for any roles but not only MeshCOP Joiners.

Do you mean we should not implement those CLI sub-commands under the `joiner` command? Do you have a recommendation of the CLI commands format?

> Do we want align the scenarios between "Get" and "Set" APIs?

Yes, they are setter-getter pair.

Reply:
@lanyuwen Changed to use a CLI command `advdata`. Please review.